src/Ladybug_SolarEnvelopeAdvanced.py

```diff
@@ -0,0 +1,160 @@
+# Solar Envelope
+# Provides an envelope (as a list of points) which represents the area in which new masses could be placed withought
+# interfering with the sun rights of surrounding buildings, given a filtered list of suns
+# By Boris Plotnikov with the assistance and guidance of Prof. Guedi Capeluto
+# pborisp@gmail.com
+# Ladybug started by Mostapha Sadeghipour Roudsari is licensed
+# under a Creative Commons Attribution-ShareAlike 3.0 Unported License.
+"""
+Use this component to generate a solar envelope for a given test surface, set of solar vectors, and context geometry that you want to ensure solar access to.  Solar envelopes are typically used to illustrate the volume that can be built within in order to ensure that a new development does not shade the surrounding properties for a given set of sun vectors.
+
+-
+Provided by Ladybug 0.0.59
+    
+    Args:
+        _baseSrf: A surface representing the area for which you want to create the solar envelope.
+        _obstacleCurves: A list of curves indicating the bottom borders of your surroundings for which you would like solar access to be kept.
+        _sunVectors: Sun vectors representing hours of the year when sun should be accessible to the properties surrounding the baseSrf.  sunVectors can be generated using the Ladybug sunPath component. 
+        _gridSize: An numeric value inidcating the gird size of the analysis in Rhino model units. The smaller the grid size - the more test points( more accurate but slower).
+        _runIt: Set to True to run the component and generate a solar envelope.
+    Returns:
+        readMe!:Log of the component
+        finalPointList: A list of points representing the heights to which you can build without shading any of the _obstacleCurves from the input _sunVectors.
+        total_ms: The time that it took this component to run
+"""
+ghenv.Component.Name = 'Ladybug_SolarEnvelopeAdvanced'
+ghenv.Component.NickName = 'SolarEnvelopeAdvanced'
+ghenv.Component.Message = 'VER 0.0.59\nAPR_20_2015'
+ghenv.Component.Category = "Ladybug"
+ghenv.Component.SubCategory = "3 | EnvironmentalAnalysis"
```

Comment:
I tend to ask everyone to put the components under WIP to be tested first and once we are sure about the performance we can move it to the main subcategory. For ladybug it is: "6 | WIP"